src/app/public/modules/auth-http/auth-interceptor.ts

```diff
@@ -0,0 +1,119 @@
+//#region imports
+
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest
+} from '@angular/common/http';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import 'rxjs/add/observable/fromPromise';
+import 'rxjs/add/operator/switchMap';
+
+import {
+  SkyAppConfig
+} from '@skyux/config';
+
+import {
+  SkyAuthTokenProvider
+} from './auth-token-provider';
+
+import {
+  SKY_AUTH_PARAM_AUTH,
+  SKY_AUTH_PARAM_PERMISSION_SCOPE
+} from './auth-interceptor-params';
+
+//#endregion
+
+function removeSkyParams(request: HttpRequest<any>): HttpRequest<any> {
+  /* istanbul ignore else */
```

Comment:
Any reason to chose this method over having a test that doesn't supply the `param` property?

Reply:
I verified that `params` is always defined, even when not specified on the original HTTP request.  I left the `if` statement and ignore comment in and added a comment above it to indicate why it's being ignored.